user_tools/src/spark_rapids_tools/tools/qualification_stats_report.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation of the Qualification Stats Report."""
+
+
+from dataclasses import dataclass, field
+from logging import Logger
+
+import pandas as pd
+import fire
+
+from spark_rapids_pytools.common.utilities import ToolLogging
+
+
+@dataclass
+class SparkQualificationStats:
+    """
+    Encapsulates the logic to generate the Qualification Stats Report.
+    """
+    logger: Logger = field(default=None, init=False)
+    unsupported_operators_file: str
+    stages_file: str
+    output_file: str
+    unsupported_operators_df: pd.DataFrame = field(default=None, init=False)
+    stages_df: pd.DataFrame = field(default=None, init=False)
+    result_df: pd.DataFrame = field(default=None, init=False)
+
+    def __init__(self, unsupported_operators_file: str, stages_file: str, output_file: str):
+        self.unsupported_operators_file = unsupported_operators_file
+        self.stages_file = stages_file
+        self.output_file = output_file
+        self.logger = ToolLogging.get_and_setup_logger('rapids.tools.qualification.stats')
+
+    def load_data(self):
+        try:
+            self.logger.info('Loading data from CSV files...')
+            self.unsupported_operators_df = pd.read_csv(self.unsupported_operators_file)
+            self.stages_df = pd.read_csv(self.stages_file)
+
+            # Convert durations from milliseconds to seconds
+            self.unsupported_operators_df['Stage Duration'] = (
+                    self.unsupported_operators_df['Stage Duration'] / 1000)
+            self.unsupported_operators_df['App Duration'] = (
+                    self.unsupported_operators_df['App Duration'] / 1000)
+            self.stages_df['Stage Task Duration'] = self.stages_df['Stage Task Duration'] / 1000
+            self.stages_df['Unsupported Task Duration'] = (
+                    self.stages_df['Unsupported Task Duration'] / 1000)
```

Comment:
nit: to be consistent with merging data, maybe we can log "Loading data complete" here?